### ISSUE TITLE
bug: reservation parity cannot repair archive-only file_reservation artifacts when global IDs collide across project archives

### ISSUE THREAD

**Issue by etafund** [Dicklesworthstone/mcp_agent_mail_rust#167]
## Summary

`am doctor health` can fail reservation parity with `archive_without_db` entries that are not safely insertable into SQLite because the same reservation IDs already exist globally under a different project. The current parity detector reports the drift but is detect-only, leaving operators to discover by hand that the archive artifacts are stale duplicates rather than missing DB rows.

This happened today during mailbox recovery: seven `file_reservations/id-*.json` artifacts existed under one project archive ("CAAM"), but the same numeric reservation IDs already existed in SQLite and in another archive for unrelated reservations.

## Observed evidence

On 2026-06-24, after clearing a deleted-executable mailbox owner, `am doctor health` still failed with:

```text
reservation_parity: drift total=7 db=14777 archive=14784 fields=[archive_without_db=7]
```

The targeted detector reported seven archive-only artifacts under:

```text
/home/ubuntu/.mcp_agent_mail_git_mailbox_repo/projects/data-projects-caam-localfix-caam-src/file_reservations/
```

Affected CAAM archive artifacts:

```text
id-13800.json
id-13815.json
id-13816.json
id-13829.json
id-13830.json
id-13863.json
id-13867.json
```

But the SQLite rows for those IDs already existed globally under another project, `data-projects-xyz`, with different agent/project ownership. Matching XYZ archive artifacts also existed. That means a naive repair that inserted the CAAM artifacts into DB would collide with existing primary keys and/or corrupt project ownership.

The CAAM artifacts were stale duplicates. Quarantining them with `git mv` inside the CAAM `file_reservations/` tree brought parity back to zero drift:

```text
reservation_parity: ok db=14777 archive=14777 drift=0
```

## Why this matters

The parity key in the archive is effectively `(project_slug, reservation_id)`, but SQLite reservation IDs are global. If old archive artifacts with globally reused IDs remain in another project directory, the detector reports `archive_without_db`, yet the correct fix is not DB insertion. The safe fix is to classify the artifact as a duplicate/global-ID collision and quarantine or rename it out of the active scan path.

This is easy for coding agents to mishandle because `archive_without_db` sounds like “insert missing DB row,” but in this case that would be wrong.

## Reproduction sketch

1. Have project A contain archive artifact `projects/project-a/file_reservations/id-123.json`.
2. Have SQLite contain reservation row `id=123` for project B, and project B contain its matching archive artifact.
3. Run reservation parity detection.
4. Observe `archive_without_db` for project A.

Expected: detector identifies `project-a/id-123.json` as a global-ID collision / duplicate active-scan artifact and recommends quarantine, not DB reconstruction.

Actual: detector reports archive-only drift but provides no repair path.

## Suggested implementation

Add a dedicated classification to reservation parity for active archive artifacts whose `id` exists in SQLite under a different project.

Agent-friendly design targets:

- Extend parity output with something like `archive_id_collision` or `archive_duplicate_global_id`.
- Include both projects in evidence: archive project slug, DB project slug, reservation ID, archive path, DB row owner/agent if available.
- Provide a safe doctor fixer that moves only the duplicate archive artifact out of the active scan path into a clearly named quarantine directory inside the same project archive. Use `Op::Rename` semantics and preserve Git auditability.
- Refuse auto-fix if the archive artifact content is newer than or otherwise materially different from the DB row without a clear duplicate signal.
- Add a regression fixture where `(project_slug, id)` exists only in archive while `id` exists in DB for a different project.

## Acceptance criteria

- `am doctor fix --only fm-db-state-files-reservation-db-archive-parity --list --json` distinguishes ordinary `archive_without_db` from cross-project global ID collisions.
- The finding evidence includes enough paths/IDs for an agent to audit without opening SQLite manually.
- A safe auto-fix or explicit repair command quarantines only duplicate active-scan artifacts and leaves canonical artifacts untouched.
- After the fixer, `am doctor health` reports `reservation_parity: ok ... drift=0`.
- Regression coverage prevents treating a global ID collision as a missing DB row to insert.

**Comment by Dicklesworthstone** [comment]
Confirmed, and the diagnosis is precise. `file_reservations.id` is a global `AUTOINCREMENT` PK, but the parity checker keys on `(project_slug, reservation_id)` (`crates/mcp-agent-mail-tools/src/reservation_parity.rs`) while archive artifacts live under `projects/<slug>/file_reservations/id-<id>.json`. So a stale `caam/id-13800.json` whose id actually belongs to project XYZ in the DB never matches at key `(caam,13800)` and falls into the `(None, Some(_))` arm → reported as `archive_without_db`, which reads like "insert the missing row" but would collide on the PK.

The current fixer is detect-only, so nothing auto-inserts (no auto-corruption), but the manual remediation text suggesting `am doctor reconstruct` is wrong for this case — confirming the mishandle risk. The DB query already returns every reservation's real `project_slug`, so a global-id→project map is available to add an `archive_id_collision` classification plus a safe rename-to-quarantine fixer (preserving git auditability), matching your acceptance criteria. Holding for the next release (workspace currently build-blocked on an unrelated dep-resolution issue).

**Comment by Dicklesworthstone** [comment]
Fixed in v0.3.15 (commit 5fdd3a8c). Reservation parity now distinguishes a cross-project global-id collision (a stale duplicate `id-<id>.json` whose reservation id is owned by another project in SQLite) from a genuine `archive_without_db` row, via a new `archive_id_collision` drift class with both project slugs in the evidence. The doctor FM `fm-db-state-files-reservation-db-archive-parity` is now auto-fixable for exactly that case: `am doctor fix --only fm-db-state-files-reservation-db-archive-parity --yes` quarantines only the duplicate archive artifacts via a reversible `Op::Rename` (`am doctor undo`), leaving canonical artifacts and all other (detect-only) drift untouched.
